File: frontend/src/features/auth/api/authApi.ts
import { http } from '@/services/http'
import type { AuthResponse } from '@/types'

export interface Credentials {
  email: string
  password: string
}

export function loginApi(payload: Credentials): Promise<AuthResponse> {
  return http.post<AuthResponse>('/api/auth/login', payload).then((r) => r.data)
}

export function registerApi(payload: Credentials): Promise<AuthResponse> {
  return http.post<AuthResponse>('/api/auth/register', payload).then((r) => r.data)
}

File: mobile/lib/api/auth_api.dart
import '../models/auth_response.dart';
import 'api_client.dart';

class Credentials {
  const Credentials({required this.email, required this.password});

  final String email;
  final String password;

  Map<String, dynamic> toJson() => {
        'email': email,
        'password': password,
      };
}

/// Communication layer for the auth endpoints:
///   - `POST /api/auth/login`
///   - `POST /api/auth/register`
class AuthApi {
  AuthApi(this._client);

  final ApiClient _client;

  Future<AuthResponse> login(Credentials credentials) async {
    final json = await _client.postJson('/api/auth/login', body: credentials.toJson());
    return AuthResponse.fromJson(json);
  }

  Future<AuthResponse> register(Credentials credentials) async {
    final json = await _client.postJson('/api/auth/register', body: credentials.toJson());
    return AuthResponse.fromJson(json);
  }
}